pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -849,27 +864,49 @@ func Run() {
 		case <-stillRunning.C:
 		}
 		agentlog.StillRunning(agentName)
+
+		//if the trigger publish device info has failed, retry
+		retryTriggerPublishDevInfo(&zedagentCtx)
+	}
+}
+
+// if trigger publish device info has failed, try again
+func retryTriggerPublishDevInfo(ctxPtr *zedagentContext) {
+	if ctxPtr.devInfoTriggerFail && ctxPtr.devInfoTaskReady {
+		log.Infof("Retrying trigger PublishDeviceInfo\n")
+		ctxPtr.devInfoTriggerFail = false
+		triggerPublishDevInfo(ctxPtr)
 	}
 }
 
 func triggerPublishDevInfo(ctxPtr *zedagentContext) {
 
+	// last publish event has to be complete,
+	// for handling current publish event trigger,
+	// wait until then
+	if ctxPtr.devInfoTriggerFail {
```

Comment:
Hmm.. Should you be checking for devInfoTaskReady here?

  if ctxPtr.devInfoTaskReady {
      return
 }

Reply:
This is avoid, queing more triggers to channel. We have a retry mechanism, when the trigger processing is over.